### PR TITLE
fix : fix frozen camera in LivePreview

### LINKED ISSUE #720: [BUG] MLKit Kotlin Sample doesn't work, camera is frozen.
MLKit sample Kotlin version has frozen camera, only classification works but only sometimes. 
Anyway, please check Kotlin version, it is not good as expect.

Java version works.

### PR DESCRIPTION
**Pull request by the-dagger**
@samtstern This should fix #720, I tested it on my local devices and the LivePreview activity in Kotlin works without issues now!

### CODE REVIEW

**Comment by the-dagger** [comment]
@samtstern Works well now, apparently the kotlinized VisionProcessorBase was the culprit here for frozen preview.
Fixing that fixed the issue for all the LivePreview demo :) 

P.S. I went ahead and made some fixes in the code to follow Kotlin style guidelines

File: mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/LivePreviewActivity.kt [hunk 1]
```diff
@@ -1,42 +1,56 @@
...
 import com.google.firebase.samples.apps.mlkit.kotlin.custommodel.CustomImageClassifierProcessor
 import com.google.firebase.samples.apps.mlkit.kotlin.facedetection.FaceContourDetectorProcessor
 import com.google.firebase.samples.apps.mlkit.kotlin.facedetection.FaceDetectionProcessor
 import com.google.firebase.samples.apps.mlkit.kotlin.imagelabeling.ImageLabelingProcessor
 import com.google.firebase.samples.apps.mlkit.kotlin.textrecognition.TextRecognitionProcessor
-import kotlinx.android.synthetic.main.activity_live_preview.facingSwitch
-import kotlinx.android.synthetic.main.activity_live_preview.fireFaceOverlay
-import kotlinx.android.synthetic.main.activity_live_preview.firePreview
-import kotlinx.android.synthetic.main.activity_live_preview.spinner
+import kotlinx.android.synthetic.main.activity_live_preview.*
```
Comment (samtstern) [inline mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/LivePreviewActivity.kt]: I guess `ktlint` allows wildcard imports now?  I had to disable them in Android Studio to get lint to pass for me.

File: mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/LivePreviewActivity.kt [hunk 2]
```diff
@@ -61,21 +75,23 @@ class LivePreviewActivity : AppCompatActivity(),
 
         setContentView(R.layout.activity_live_preview)
 
-        if (firePreview == null) {
+        preview = firePreview
```
Comment (samtstern) [inline mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/LivePreviewActivity.kt]: Can you explain why you dropped the synthetic binding and moved to use assignment to a `private var` here?

File: mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/LivePreviewActivity.kt [hunk 3]
```diff
@@ -61,21 +75,23 @@ class LivePreviewActivity : AppCompatActivity(),
...
+        if (graphicOverlay == null) {
             Log.d(TAG, "graphicOverlay is null")
         }
 
-        val options = arrayListOf(
-                FACE_CONTOUR,
+        val spinner = spinner
+        val options = arrayListOf(FACE_CONTOUR,
                 FACE_DETECTION,
-                TEXT_DETECTION,
```
Comment (samtstern) [inline mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/LivePreviewActivity.kt]: Why remove `TEXT_DETECTION` here?

File: mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/VisionProcessorBase.kt [hunk 4]
```diff
@@ -49,12 +49,8 @@ abstract class VisionProcessorBase<T> : VisionImageProcessor {
...
     // Bitmap version
     override fun process(bitmap: Bitmap, graphicOverlay: GraphicOverlay) {
-        detectInVisionImage(
-            null /* bitmap */,
-            FirebaseVisionImage.fromBitmap(bitmap),
-            null,
-            graphicOverlay
-        )
+        detectInVisionImage(null, FirebaseVisionImage.fromBitmap(bitmap), null,
+                graphicOverlay)/* bitmap */
```
Comment (samtstern) [inline mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/VisionProcessorBase.kt]: The `/* bitmap */` comment at the end of this line seems misplaced.

File: mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/facedetection/FaceContourDetectorProcessor.kt [hunk 5]
```diff
@@ -59,12 +42,32 @@ class FaceContourDetectorProcessor : VisionProcessorBase<List<FirebaseVisionFace
...
+    }
+
     override fun onFailure(e: Exception) {
         Log.e(TAG, "Face detection failed $e")
     }
 
     companion object {
-
-        private const val TAG = "FaceContourDetectorProc"
+        private val TAG = "FaceContourDetectorProc"
```
Comment (samtstern) [inline mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/facedetection/FaceContourDetectorProcessor.kt]: Why is this not `const`?

File: mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/VisionProcessorBase.kt [hunk 6]
```diff
@@ -100,22 +89,27 @@ abstract class VisionProcessorBase<T> : VisionImageProcessor {
         graphicOverlay: GraphicOverlay
     ) {
         detectInImage(image)
-            .addOnSuccessListener { results ->
-                metadata?.let {
-                    onSuccess(originalCameraImage!!, results, it, graphicOverlay)
+                .addOnSuccessListener { results ->
+                    this@VisionProcessorBase.onSuccess(originalCameraImage, results,
```
Comment (samtstern) [inline mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/VisionProcessorBase.kt]: Was this what was causing the freeze?

File: mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/VisionProcessorBase.kt [hunk 7]
```diff
@@ -100,22 +89,27 @@ abstract class VisionProcessorBase<T> : VisionImageProcessor {
...
+                            metadata!!,
+                            graphicOverlay)
+                    processLatestImage(graphicOverlay)
```
Comment (the-dagger) [inline mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/VisionProcessorBase.kt]: @samtstern This was what's causing the error and hence as a result only the first frame of the preview was displayed.

File: mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/LivePreviewActivity.kt (hunk 1 above)
Comment (the-dagger) [reply mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/LivePreviewActivity.kt]: Yeah I think.
I'll remove the wildcard import and replace it with specific ones

File: mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/VisionProcessorBase.kt (hunk 7 above)
Comment (samtstern) [reply mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/VisionProcessorBase.kt]: Got it, that makes sense

File: mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/LivePreviewActivity.kt (hunk 2 above)
Comment (the-dagger) [reply mlkit/app/src/main/java/com/google/firebase/samples/apps/mlkit/kotlin/LivePreviewActivity.kt]: I essentially reconverted this file again to ensure that nothin was wrong here, so it seems like I missed some issues. 
Will fix these too

**Comment by samtstern** [comment]
@the-dagger 
```
./build.sh: line 34: /Users/harshitdwivedi/Library/Android/sdk/tools/bin/sdkmanager: No such file or directory
```

Build failure implies some of your local setup snuck into this PR.

**Comment by the-dagger** [comment]
Accidental commit 🤦‍♂️ 
I edited that file as I was running it locally to ensure that the travis test will pass.
Looks like being extra careful can bite you in the back too! 😆

**Comment by samtstern** [comment]
@the-dagger thanks for solving this one!

**Comment by the-dagger** [comment]
Anytime!

**Pull request merged**